### ISSUE TITLE
Adding functions to Fortran APIs means only latest netcdf-c will build netcdf-fortran...

### ISSUE THREAD

**Issue by edwardhartnett** [Unidata/netcdf-fortran#215]
I have been working on adding nc_def_var_szip() to the Fortran APIs. This is not the only new function, @DennisHeimbigner has some new ones as well that will have to be supported in netcdf-fortran.

The last time that a new function was added was nc_def_var_filter/nc_inq_var_filter. When these were added there was a discontinuity in how to build netcdf-fortran. netcdf-fortran can only be built with versions of netcdf-c 4.6.0 or greater, otherwise the build will fail with undefined references.

The same will be true in this case. When Dennis and I finish adding new functions, netcdf-fortran will only compile with the most recent netcdf-c code. 

Perhaps this was already obvious to everyone but me, but I wanted to make it clear. ;-)

I will add a check to the configure/CMake builds to inform the user at configure-time, instead of failing with mysterious link errors. (I see that this is already a desired change, see #68 )

**Comment by zerothi** [comment]
What about including the netcdf.h and check the versions there? Possibly adding a netcdf_version.h in C? This has always been a problem for fortran and I see no good solution other than what I propose?

Also this will go perfectly with the .F90 extension to allow preprocessor statements.

**Comment by edwardhartnett** [comment]
Detecting the condition is not a problem. I can look for nc_def_var_szip() and if it is not present, then netcdf-c is too old.

This could be handled with pre-processor but that is not how we have done this in the past. None of our current code uses pre-processor statements, because it was all written long ago, and the pre-processor is not formally part of the standard.

I am willing to pursue the pre-processor solution, but I would suggest that be done after this release. It will take some experimentation and testing to ensure pre-processing is working on every platform (including windows). All the Fortran compilers I'm aware of can handle it, but I only have gfortran and ifort to test with.

In other words, we should do this release, with the new function support, and that will require the latest netcdf-c code. Then, experiments in pre-processing can begin, and next time we need to add functions, we could use the pre-processor to handle it.

**Comment by zerothi** [comment]
Sorry, I meant during the compilation of the netcdf-fortran library. Since that way you would not need a specific netcdf-c version. *If* the number of functions grows continuously, it would be easier to simply check during compilation and remove the function (or provide a dummy function which emits an error?).
